### PR TITLE
Restructure api and add event-status model

### PR DESCRIPTION
**Pull request by toddapetersen**
This is a baseline restructure of the Agency api to address a number of endpoint changes and incorporate feedback received in previous issue and PRs (#161 , #116  and #63 ).

Major changes:
* `provider_id` is passed as part of an authorization bearer token.
* Looking up and registering vehicles use standard `GET` and `POST` operations on `/vehicles` endpoint.
* The `/vehicles/{vehicle_id}/event` allows Providers to `POST` events to affect vehicle status.  The vehicle status is modeled around a state flow that is modeled in Vehicle Events table.
* Deregister vehicle, trip start and end are now events.
* Events for `trip_enter` and `trip_leave` were added to address trips that do not start or end in the Agency jurisdiction.

### CODE REVIEW

**Comment by HenriJ** [comment]
Hi Todd, love the more restful approach !
* Why send telemetry data in the {vehicle}/event endpoint instead of the /update_trip_telemetry (that could become a {vehicle}/telemetry) ? It might be interesting to be updated on the telemetry every 5 seconds (to keep up-to-date data on the battery level, ...).
* The `Telemetry` is referenced in the document, but I can't find the definition ?

**Comment by HenriJ** [comment]
Oops, just saw your PR https://github.com/CityOfLosAngeles/mobility-data-specification/pull/196 , it answers my questions :)

For the events, what should happen if the Agency server goes down ? Could we add a timestamp field the event so that if we receive it later, we can know the real time when it occurred ?

**Comment by hunterowens** [comment]
@toddapetersen 

Couple things. 

This PR needs to target the `dev` branch, not `0.2.x` as it is breaking changes. 

I noticed that #195 #196 #197 all target this branch rather than `dev` or `0.2.x` and that is PR contains commits that should be in that are also in #195/#196/#197? Or do you have a different strategy to merge these commits? I think it is currently very unclear how changes are being made.

**Comment by hunterowens** [comment]
Need to additionally integrate rebase and integrate #172 before merging this.

File: agency/README.md [hunk 1]
```diff
@@ -1,136 +1,106 @@
...
-| ----- | -------- | ----------------- | ----- |
-| `message` | Enum |  | See [Message](#message) Enum |
+| Field         | Type           | Field Description                                                             |
+| ------------- | -------------- | ----------------------------------------------------------------------------- |
+| `vehicle_id`  | UUIDv4         | Provided by Operator to uniquely identify a vehicle                           |
+| `provider_id` | UUIDv4         | Issued by <insert here>                                                       |
+| `vin`         | String         | Vehicle Identification Number (VIN) visible on vehicle                        |
+| `type`        | Enum           | [Vehicle Type](#vehicle-type)                                                 |
+| `propulsion`  | Enum[]         | Array of [Propulsion Type](#propulsion-type); allows multiple values          |
+| `year`        | Integer        | Year Manufactured                                                             |
```
[the remark this answers is not in the document]
Comment (toddapetersen) [reply agency/README.md]: Tag these params as optional

File: agency/README.md [hunk 2]
```diff
@@ -1,136 +1,106 @@
...
-| `event_type` | Enum | Required | [Event Type](#event_type) for status change.  |
-| `reason_code` | Enum | Required | [Reason](#reason_code) for status change.  |
-| `battery_pct` | Float | Require if Applicable | Percent battery charge of device, expressed between 0 and 1 |
+| Field        | Type    | Required/Optional | Field Description                                                    |
+| ------------ | ------- | ----------------- | -------------------------------------------------------------------- |
+| `vehicle_id` | UUIDv4  | Required          | Provided by Operator to uniquely identify a vehicle                  |
+| `vin`        | String  | Required          | Vehicle Identification Number (VIN) visible on vehicle               |
+| `type`       | Enum    | Required          | [Vehicle Type](#vehicle-type)                                        |
+| `propulsion` | Enum[]  | Required          | Array of [Propulsion Type](#propulsion-type); allows multiple values |
+| `year`       | Integer | Required          | Year Manufactured                                                    |
```
[the remark this answers is not in the document]
Comment (toddapetersen) [reply agency/README.md]: change to vehicle_id as current vehicle_id becomes device_id.
```suggestion
| `year`       | Integer | Required          | Year Manufactured                                                    |
```

File: agency/README.md [hunk 3]
```diff
@@ -1,136 +1,106 @@
...
 * [update_trip_telemetry](#update_trip_telemetry)
 * [service_areas](#service_areas)
 * [Event types](#Event-Types)
 * [Enum definitions](#enum-definitions)
 
-## register_vehicle
+## Authorization
 
-The Vehicle Registration API is required in order to register a vehicle for use in the system. The API will require a valid `provider_id` and `api_key`.
+When making requests, the Agency API expects `provider_id` to be part of the claims in a [JWT](https://jwt.io/)  `access_token` in the `Authorization` header, in the form `Authorization: Bearer <access_token>`. The token issuance, expiration and revocation policies are at the discretion of the Agency.
```
[the remark this answers is not in the document]
Comment (toddapetersen) [reply agency/README.md]: JWT bearer token will be accepted from a trusted identity provider and include `provider_id` as a token claim.

File: agency/README.md [hunk 4]
```diff
@@ -1,136 +1,106 @@
...
+| `mfgr`       | String  | Required          | Vehicle Manufacturer                                                 |
+| `model`      | String  | Required          | Vehicle Model                                                        |
 
-Response:
+201 Success Response:
 
-| Field | Type     | Required/Optional | Other |
-| ----- | -------- | ----------------- | ----- |
-| `message` | Enum |  | See [Message](#message) Enum |
+_No content returned on success._
```
[the remark this answers is not in the document]
Comment (toddapetersen) [reply agency/README.md]: Resolve in #195 .

**Comment by toddapetersen** [comment]
Discussed changes are committed and rebased against dev.  Should be good to merge.

**Pull request merged**